presentation/src/main/java/com/anytypeio/anytype/presentation/search/GlobalSearchViewModel.kt

```diff
@@ -55,47 +63,143 @@ class GlobalSearchViewModel(
     private val urlBuilder: UrlBuilder
 ) : BaseViewModel() {
 
-    private val userInput = MutableStateFlow("")
+    private val userInput = MutableStateFlow(EMPTY_STRING_VALUE)
     private val searchQuery = userInput
         .take(1)
         .onCompletion {
-            emitAll(userInput.drop(1).debounce(ObjectSearchViewModel.DEBOUNCE_DURATION).distinctUntilChanged())
+            emitAll(userInput.drop(1).debounce(DEFAULT_DEBOUNCE_DURATION).distinctUntilChanged())
         }
 
+    private val mode = MutableStateFlow<Mode>(Mode.Default)
+
     val views = MutableStateFlow<List<GlobalSearchItemView>>(emptyList())
     val navigation = MutableSharedFlow<OpenObjectNavigation>()
 
-    init {
-        viewModelScope.launch {
-            searchQuery
-                .flatMapLatest { query ->
-                    searchWithMeta
-                        .asFlow(
-                            Command.SearchWithMeta(
-                                query = query,
-                                limit = 50,
-                                offset = 0,
-                                keys = emptyList(),
-                                filters = ObjectSearchConstants.filterSearchObjects(
-                                    // TODO add tech space?
-                                    spaces = listOf(spaceManager.get())
-                                ),
-                                sorts = ObjectSearchConstants.sortsSearchObjects,
-                                withMetaRelationDetails = true,
-                                withMeta = true
-                            )
-                        )
-                }.collect { results ->
-                    views.value = results.mapNotNull { result ->
-                        result.view(
-                            storeOfObjectTypes = storeOfObjectTypes,
-                            storeOfRelations = storeOfRelations,
-                            urlBuilder = urlBuilder
+    val state = combine(
+        mode,
+        searchQuery
+    ) { mode, query ->
+        mode to query
+    }.flatMapLatest { (mode, query) ->
+        when(mode) {
+            is Mode.Default -> {
+                buildDefaultSearchFlow(query)
+            }
+            is Mode.Related -> {
+                buildRelatedSearchFlow(query, mode)
+            }
+        }
+    }.stateIn(
+        scope = viewModelScope,
+        started = SharingStarted.WhileSubscribed(),
+        initialValue = ViewState.Init
+    )
+
+    private fun buildRelatedSearchFlow(
+        query: String,
+        mode: Mode.Related
+    ) = searchWithMeta
+        .stream(
+            Command.SearchWithMeta(
+                query = query,
+                limit = 50,
+                offset = 0,
+                keys = emptyList(),
+                filters = buildList {
+                    add(
```

Comment:
it's better to have all filters inside ObjectSearchConstants class

Reply:
done